Core/StarlightItem.Events.cs

```diff
@@ -4,6 +4,20 @@ namespace StarlightRiver.Core
 {
 	internal partial class StarlightItem : GlobalItem
 	{
+		public delegate void ExtractinatorUseDelegate(int extractType, int extractinatorBlockType, ref int resultType, ref int resultStack);
+		public static event ExtractinatorUseDelegate ExtractinatorUseEvent;
+		public override void ExtractinatorUse(int extractType, int extractinatorBlockType, ref int resultType, ref int resultStack)
+		{
+			ExtractinatorUseEvent?.Invoke(extractType, extractinatorBlockType, ref resultType, ref resultStack);
+		}
+
+		public delegate void SetStaticDefaultsDelegate();
+		public static event SetStaticDefaultsDelegate SetStaticDefaultsEvent;
+		public override void SetStaticDefaults()
```

Comment:
shimmer recipes. on second thought it could probably just be placed in the mod item's setstaticdefaults but i wanted to keep consistency as if i were to do it with a global item

Reply:
is there a reason for these to be enforced as a global item though? Wpuld it not make more sense to have relevant recipes in an items own class, and then handle custom registration for vanilla items seperately?